### PR TITLE
Converted most of the tests to pytest

### PR DESCRIPTION
**Pull request by lechat**
Not all test converted yet, but it gets hard to re-adapt tests with each pull request. As most of the tests already converted and pytest have no problems running existing unittest based tests, I decided to merge this work to master.

### CODE REVIEW

File: jenkinsapi/jenkinsbase.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __eq__(self, other):
         """
         if not isinstance(other, self.__class__):
             return False
-        if not other.baseurl == self.baseurl:
+        if other.baseurl != self.baseurl:
```
Comment (bmwant) [inline jenkinsapi/jenkinsbase.py]: return `other.baseurl == self.baseurl`

File: jenkinsapi_tests/systests/test_credentials.py [hunk 2]
```diff
@@ -18,147 +13,151 @@
...
-        cred_dict = {
-            'description': cred_descr,
-            'userName': 'userName',
-            'passphrase': '',
-            'private_key': '-----BEGIN RSA PRIVATE KEY-----'
-        }
-        creds[cred_descr] = SSHKeyCredential(cred_dict)
+def test_get_credentials(jenkins):
+    creds = jenkins.credentials
+    assert isinstance(creds, Credentials) is True
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_credentials.py]: `assert isinstance` is enough, remove `is True`

File: jenkinsapi_tests/systests/test_credentials.py [hunk 3]
```diff
@@ -18,147 +13,151 @@
...
 
-        self.assertTrue(cred_descr in creds)
-        cred = creds[cred_descr]
-        self.assertIsInstance(cred, SSHKeyCredential)
-        self.assertEquals(cred.description, cred_descr)
+    assert cred_descr in creds
 
-        del creds[cred_descr]
+    cred = creds[cred_descr]
+    assert isinstance(cred, SSHKeyCredential) is True
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_credentials.py]: same

File: jenkinsapi_tests/systests/test_credentials.py [hunk 4]
```diff
@@ -18,147 +13,151 @@
...
+        'description': cred_descr,
+        'userName': 'userName',
+        'passphrase': '',
+        'private_key': '/tmp/key'
+    }
+    creds[cred_descr] = SSHKeyCredential(cred_dict)
+
+    assert cred_descr in creds
+    cred = creds[cred_descr]
+    assert isinstance(cred, SSHKeyCredential) is True
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_credentials.py]: same

File: jenkinsapi_tests/systests/test_credentials.py [hunk 5]
```diff
@@ -18,147 +13,151 @@
...
+        'description': cred_descr,
+        'userName': 'userName',
+        'passphrase': '',
+        'private_key': '~/.ssh/key'
+    }
+    creds[cred_descr] = SSHKeyCredential(cred_dict)
+
+    assert cred_descr in creds
+    cred = creds[cred_descr]
+    assert isinstance(cred, SSHKeyCredential) is True
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_credentials.py]: same

File: jenkinsapi_tests/systests/test_crumbs_requester.py [hunk 6]
```diff
@@ -22,89 +17,82 @@
...
     },
     'core:apply': ''
-  }
+}
+
 
+@pytest.fixture(scope='function')
+def crumbed_jenkins(jenkins):
+    jenkins.requester.post_and_confirm_status(
+        jenkins.baseurl + '/configureSecurity/configure',
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_crumbs_requester.py]: you can use `urljoin` to be sure that result will not contain extra slashes (e.g. if baseurl ends with one)

File: jenkinsapi_tests/systests/test_crumbs_requester.py [hunk 7]
```diff
@@ -22,89 +17,82 @@
...
 
-        job_name = 'create1_%s' % random_string()
-        job = self.jenkins.create_job(job_name, JOB_WITH_FILE)
+    job_name = 'create1_%s' % random_string()
+    job = crumbed_jenkins.create_job(job_name, JOB_WITH_FILE)
 
-        self.assertTrue(job.has_params())
-        self.assertTrue(len(job.get_params_list()) != 0)
+    assert job.has_params() is True
+    assert len(job.get_params_list()) != 0
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_crumbs_requester.py]: both just `assert` without `is True` and `!= 0`

File: jenkinsapi_tests/systests/test_invocation.py [hunk 8]
```diff
@@ -20,104 +15,106 @@
...
 
-    def test_multiple_invocations_and_get_last_build(self):
-        job_name = 'Dcreate_%s' % random_string()
+def test_get_block_until_build_running(jenkins):
+    job_name = 'Bcreate_%s' % random_string()
+    job = jenkins.create_job(job_name, LONG_RUNNING_JOB)
+    qq = job.invoke()
+    time.sleep(3)
+    bn = qq.block_until_building(delay=3).get_number()
+    assert isinstance(bn, int) is True
```
Comment (bmwant) [inline jenkinsapi_tests/systests/test_invocation.py]: same

File: jenkinsapi_utils/jenkins_launcher.py [hunk 9]
```diff
@@ -64,9 +74,16 @@ class JenkinsLancher(object):
...
-        self.jenkins_url = jenkins_url
-        self.http_port = random.randint(9000, 10000) if not jenkins_url \
-            else urlparse(jenkins_url).port
+        if jenkins_url is not None:
+            self.jenkins_url = jenkins_url
+            self.http_port = urlparse(jenkins_url).port
+            self.start_new_instance = False
+        else:
+            self.http_port = random.randint(9000, 10000)
+            self.jenkins_url = 'http://localhost:%s' % self.http_port
```
Comment (bmwant) [inline jenkinsapi_utils/jenkins_launcher.py]: try `sock.bind(('', 0))` and then `sock.getsockname()[1]` to get random port selected by os that guaranteed to be not in use

**Comment by lechat** [comment]
This is taking too long. I will merge it now even is some tests are still unittest style.

**Pull request merged**